cypress-tests/cypress/e2e/editor/widget/buttonHappyPath.cy.js

```diff
@@ -1,100 +1,344 @@
+    cy.verifyToastMessage(commonSelectors.toastMessage, data.alertMessage);
+
+    openEditorSidebar(data.widgetName);
+    openAccordion(commonWidgetText.accordionGenaral);
+    addAndVerifyTooltip(
+      commonWidgetSelector.draggableWidget(data.widgetName),
+      data.customMessage
+    );
+
+    verifyLayout(data.widgetName);
+
+    cy.get(commonWidgetSelector.changeLayoutButton).click();
+    cy.get(
+      commonWidgetSelector.parameterTogglebutton(
+        commonWidgetText.parameterShowOnDesktop
+      )
+    ).click();
+
+    cy.get(commonWidgetSelector.widgetDocumentationLink).should(
+      "have.text",
+      buttonText.buttonDocumentationLink
+    );
+
+    verifyControlComponentAction(data.widgetName, data.customMessage);
+
+    cy.get(commonSelectors.editorPageLogo).click();
+    cy.deleteApp(data.appName);
   });
 
-  it("should verify the styles of the button widget",()=>{
-   button.stylesElements();
-   button.colorPickerCard();
-   cy.get(buttonSelector.stylesFx.visibility).click();
-   cy.get(buttonSelector.fxVisibility).should("have.text",buttonText.trueText);
-   cy.get(buttonSelector.visibilityCloseFx).click();
-   cy.get(buttonSelector.visibilityToggle).uncheck();
-   cy.get(buttonSelector.stylesFx.visibility).click();
-   cy.get(buttonSelector.fxVisibility).should("have.text",buttonText.falseText);
-   cy.get(buttonSelector.visibilityCloseFx).click();
-   cy.get(buttonSelector.buttonWidget).should("not.be.visible");
-   cy.get(buttonSelector.visibilityToggle).check();
-
-   cy.get(buttonSelector.stylesFx.disable).click();
-   cy.get(buttonSelector.fxDisable).should("have.text",buttonText.falseText);
-   cy.get(buttonSelector.disableCloseFx).click();
-   cy.get(buttonSelector.disableToggle).check();
-   cy.get(buttonSelector.stylesFx.disable).click();
-   cy.get(buttonSelector.fxDisable).should("have.text",buttonText.trueText);
-   cy.get(buttonSelector.disableCloseFx).click();
-   cy.get(buttonSelector.buttonWidget).should("be.disabled");
-   cy.get(buttonSelector.disableToggle).uncheck();
-   cy.get(buttonSelector.buttonWidget).should("be.enabled");
-
-   cy.get(buttonSelector.stylesInput.borderRadiusInputField).clear().type("15");
-   cy.get(buttonSelector.stylesFx.borderRadius).click();
-   cy.get(buttonSelector.fxBorderRadius).should("have.text", buttonText.borderRadiusInput)
-
-   button.deleteApp();
+  it("should verify the styles of the button widget", () => {
+    const data = {};
+    data.appName = `${fake.companyName}-App`;
+    data.colourHex = fake.randomRgbaHex;
+    data.colour = fake.randomRgba;
+    data.boxShadowParam = fake.boxShadowParam;
+
+    cy.renameApp(data.appName);
+
+    openEditorSidebar(buttonText.defaultWidgetName);
+    cy.get(commonWidgetSelector.buttonStylesEditorSideBar).click();
+
+    verifyAndModifyStylePickerFx(
+      buttonText.backgroundColor,
+      buttonText.defaultBackgroundColor,
+      data.colourHex
```

Comment:
I think we can use a different color for each color field.
We can assign different values for `data.colourHex`